dev-ostelco-ios-clientTests/CoordinatorTests/RootCoordinatorTests.swift

```diff
@@ -0,0 +1,167 @@
+//
+//  RootCoordinatorTests.swift
+//  dev-ostelco-ios-clientTests
+//
+//  Created by Ellen Shapiro on 6/5/19.
+//  Copyright © 2019 mac. All rights reserved.
+//
+
+import ostelco_core
+@testable import Oya_Development_app
+import XCTest
+
+class RootCoordinatorTests: XCTestCase {
+    
+    lazy var testUserManager = MockUserManager()
+    
+    lazy var testCoordinator = RootCoordinator(window: UIWindow(),
+                                               userManager: self.testUserManager)
+    
+    override func setUp() {
+        super.setUp()
+        UserDefaultsWrapper.clearAll()
+        APIManager.shared.primeAPI = self.mockAPI
+        self.setupStubbing()
+    }
+    
+    override func tearDown() {
+        self.tearDownStubbing()
+        APIManager.shared.primeAPI = PrimeAPI(baseURL: APIManager.shared.baseURLString, tokenProvider: UserManager.shared)
+        UserDefaultsWrapper.clearAll()
+        super.tearDown()
+    }
+    
+    func testNoLoggedInUserKicksToLogin() {
+        self.testUserManager.mockHasCurrentUser = false
+        
+        guard let destination = self.testCoordinator.determineDestination().awaitResult(in: self) else {
```

Comment:
awaitResult works without needing to use a test expectation to wait? handy

Reply:
It uses expectations [under the hood](https://github.com/ostelco/ostelco-ios-client/blob/develop/dev-ostelco-ios-clientTests/PromiseKit%2BTesting.swift#L14)

Reply:
Oh nice.